Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -0,0 +1,149 @@
+        managedObjectContext.perform {
+            let predicate = NSPredicate(format: "identifier = %@", identifier.value)
+            do {
+                let fetchRequest: NSFetchRequest<StoredAlert> = StoredAlert.fetchRequest()
+                fetchRequest.predicate = predicate
+                fetchRequest.sortDescriptors = [ NSSortDescriptor(key: "modificationCounter", ascending: true) ]
+                let result = try self.managedObjectContext.fetch(fetchRequest)
+                completion(.success(result.last))
+            } catch {
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    private func recordUpdateOfLatest(of identifier: DeviceAlert.Identifier, at timestamp: Date = Date(),
+                                      with block: @escaping (StoredAlert) -> Void,
+                                      completion: ((Result<Void, Error>) -> Void)?) {
+        dataAccessQueue.async {
+            self.managedObjectContext.perform {
+                self.lookupLatest(identifier: identifier) {
+                    switch $0 {
+                    case .success(let object):
+                        if let object = object {
+                            block(object)
+                            do {
+                                try self.managedObjectContext.save()
+                                self.log.default("Recorded alert: %{public}@", identifier.value)
+                                completion?(.success)
+                            } catch {
+                                self.log.error("Could not store alert: %{public}@", String(describing: error))
+                                completion?(.failure(error))
+                            }
+                        } else {
+                            completion?(.failure(AlertStoreError.notFound))
+                        }
+                    case .failure(let error):
+                        completion?(.failure(error))
+                    }
+                }
+            }
+        }
+    }
+    
+    public func recordAcknowledgement(of identifier: DeviceAlert.Identifier, at timestamp: Date = Date(),
+                                      completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, at: timestamp, with: { $0.acknowledgedTimestamp = timestamp }, completion: completion)
+    }
+    
+    public func recordRetraction(of identifier: DeviceAlert.Identifier, at timestamp: Date = Date(),
+                                 completion: ((Result<Void, Error>) -> Void)? = nil) {
+        recordUpdateOfLatest(of: identifier, at: timestamp, with: { $0.retractedTimestamp = timestamp }, completion: completion)
+    }
+}
+
+// MARK: Query Support
+
+extension AlertStore {
+
+    // At the moment, this is only used for unit testing
+    func fetch(identifier: DeviceAlert.Identifier, completion: @escaping (Result<[StoredAlert], Error>) -> Void) {
```

Comment:
[nit] `fileprivate` so outsiders don't use fetch independently? (I'm dealing with this in the other stores while trying to track down all uses of the store. Restricting direct access to store through limited number of intentional public APIs is a good thing.)

Reply:
Unless of course, outsiders should be using `fetch`?

Reply:
See the comment above the function.  I wish there was a way to restrict access only to unit tests, but the best we can do to make it `internal` (the default) and not `public`.  I will add an explicit `internal` keyword to make this even clearer.

Reply:
If possible, consider moving this extension into the test file itself? If you use an `@testable` import, I *think* you should be able to access to the internal properties (maybe?).